### PR TITLE
Update VITS modeling to enable ONNX export

### PR DESCRIPTION
**Pull request by echarlaix**
This PR enables the ONNX export of VITS models in Optimum (https://github.com/huggingface/optimum/pull/1607), currently the export is failing due to [a cast operator added before the pow operator](https://github.com/pytorch/pytorch/blob/v2.1.2/torch/onnx/symbolic_opset9.py#L3382) in the model graph, resulting in an issue during the concatenation of two values of different data type

cc @xenova

### CODE REVIEW

File: src/transformers/models/vits/modeling_vits.py [hunk 1]
```diff
@@ -1022,7 +1022,7 @@ def _absolute_position_to_relative_position(self, x):
 
         # Pad along column
         x = nn.functional.pad(x, [0, length - 1, 0, 0, 0, 0])
-        x_flat = x.view([batch_heads, length**2 + length * (length - 1)])
+        x_flat = x.view([batch_heads, length * length + length * (length - 1)])
```
Comment (echarlaix) [inline src/transformers/models/vits/modeling_vits.py:1025]:
```suggestion
        x_flat = x.view([batch_heads, length * (2 * length - 1)])
```
(might be clearer)

**Review by ArthurZucker — APPROVED** [review]
LGTM with the added suggestion. 
For futur reference best practice is this only when it power of 2 ? does pow(x,2) work better (but is slower I think)

**Review by xenova — APPROVED** [review]
Can confirm this fixes the ONNX exports (see [here](https://huggingface.co/models?library=transformers.js&other=vits) for a list I have converted). 👍 

We can also update [this](https://github.com/huggingface/transformers/pull/28141/files#r1431667596) for readability.

**Comment by echarlaix** [comment]
> For futur reference best practice is this only when it power of 2 ? does pow(x,2) work better (but is slower I think)

In all cases it'll be [casted to fp32](https://github.com/pytorch/pytorch/blob/v2.1.2/torch/onnx/symbolic_opset9.py#L3386) (no matter the exponent), using a [multiplication](https://github.com/onnx/onnx/blob/main/docs/Operators.md#mul) instead removes this constraint, and should be "faster" (not that it'll have any impact here)

**Pull request merged**